### PR TITLE
Dark mode

### PR DESCRIPTION
**Pull request by grahambrown11**
## Description:

Implement Dark Mode for Docs 😎 

Added an optional column on the end of the `imgtable::` for invert the image color - add `dark-invert`

## Checklist:

  - [ ] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [x] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: _static/custom.css [hunk 1]
```diff
@@ -133,11 +141,10 @@ div.index-shields {
 }
 
 .guide-card {
-  color: #004b6b;
   display: block;
   height: auto;
   width: 300px;
-  background-color: #f3f6f6;
+  background-color: #d5f7f7;
```
Comment (jesserockz) [inline _static/custom.css]: Not sure I like this new colour on these cards

File: _static/custom.css [hunk 2]
```diff
@@ -170,23 +181,141 @@ div.index-shields {
...
+    color: #ececec;
+    border-bottom: 1px dotted #ececec;
+  }
+
+  div.sphinxsidebar a:hover, .breadcrumbs a:hover {
+    border-bottom: 1px solid #ececec;
+  }
+
+  .guide-card {
+    background-color: #14504f;
```
Comment (jesserockz) [inline _static/custom.css]: I think keeping the guide just a lighter grey in dark mode fits better
```suggestion
    background-color: #393939;
```

File: conf.py [hunk 3]
```diff
@@ -100,7 +100,7 @@
 # default_role = 'cpp:any'
 
 # The name of the Pygments (syntax highlighting) style to use.
-pygments_style = "xcode"
+pygments_style = "solarized-dark"
```
Comment (jesserockz) [inline conf.py]: I think this makes the text too light in light mode

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: _static/custom.css (hunk 1 above)
Comment (grahambrown11) [reply _static/custom.css]: OK, I thought the green from the chevrons in the picture was the inspiration for the original greyish green... to make it tie in with the dark mode I made it more green, but happy to leave it more grey

File: conf.py (hunk 3 above)
Comment (grahambrown11) [reply conf.py]: That was the only one I found that more or less works for both light and dark... I see Sphinx supports `pygments_dark_style` but not as a user setting, so I'll look at creating a custom theme that extends `alabaster` so we can use `xcode` for light and `solarized-dark` (or I prefer github-dark)

**Review by jesserockz — APPROVED** [review]

**Pull request merged**